mir_eval/multipitch.py

```diff
@@ -0,0 +1,383 @@
+    target_times : np.ndarray
+        Array of target time stamps
+
+    Returns
+    -------
+    frequencies_resampled : list of lists
+        Frequency list of lists resampled to new timebase
+    """
+    n_times = len(frequencies)
+
+    frequency_index = np.arange(0, n_times)
+    new_frequency_index = interp1d(
+        times, frequency_index, kind='nearest', bounds_error=False,
+        assume_sorted=True, fill_value=n_times)(target_times)
+
+    freq_vals = frequencies + [np.array([])]
+    frequencies_resampled = [
+        freq_vals[i] for i in new_frequency_index.astype(int)]
+
+    return frequencies_resampled
+
+
+def frequencies_to_logscale(frequencies):
+    """Converts frequencies to semitone log scale.
+
+    Parameters
+    ----------
+    frequencies : list of np.ndarrays
+        Original frequency values
+
+    Returns
+    -------
+    frequencies_logscale : list of np.ndarrays
+        Frequency values in semitone log scale.
+    """
+    return [12.0*np.log2(freqs) for freqs in frequencies]
+
+
+def logscale_to_single_octave(frequencies_logscale):
+    """Wrap log scale frequencies to a single octave.
+
+    Parameters
+    ----------
+    frequencies_logscale : list of np.ndarrays
+        Log scale frequency values
+
+    Returns
+    -------
+    frequencies_logscale_chroma : list of np.ndarrays
+        Log scale frequency values wrapped to one octave.
+
+    """
+    return [np.mod(freqs, 12) for freqs in frequencies_logscale]
+
+
+def compute_num_freqs(frequencies):
+    """Computes the number of frequencies for each time point.
+
+    Parameters
+    ----------
+    frequencies : list of np.ndarrays
+        Frequency values
+
+    Returns
+    -------
+    num_freqs : np.ndarray
+        Number of frequencies at each time point.
+    """
+    return np.array([f.size for f in frequencies])
+
+
+def compute_num_true_positives(ref_freqs, est_freqs, window=0.5):
+    """Compute the number of true positives in an estimate given a reference.
+    A frequency is correct if it is within a semitone of the
+    correct frequency.
+
+    Parameters
+    ----------
+    ref_freqs : list of np.ndarrays
+        reference frequencies in semitones
+    est_freqs : list of np.ndarrays
+        estimated frequencies in semitones
+    window : float
+        Window size, in semitones
+
+    Returns
+    -------
+    true_positives : np.ndarray
+        Array the same length as ref_freqs containing the number of true
+        positives.
+
+    """
+    n_frames = len(ref_freqs)
+    true_positives = np.zeros((n_frames, ))
+
+    for i, (ref_frame, est_frame) in enumerate(zip(ref_freqs, est_freqs)):
+        if len(ref_frame) == 0 or len(est_frame) == 0:
+            true_positives[i] = 0.0
+        else:
+            matching = util.match_events(ref_frame, est_frame, window)
+            true_positives[i] = len(matching)
+
+    return true_positives
+
+
+def accuracy(true_positives, n_ref, n_est):
```

Comment:
> Is the disadvantage just that compute_num_true_positives would have to be computed twice?

yes, it could in theory be quite slow, especially if used on a full track. It's essentially performing an onset detection eval for every time stamp (albeit over a smaller number of "onsets" each time).

> Can we just combine the two metrics?

Sure. I'm a bit confused though. In ```mir_eval/melody.py``` the metrics are all written as separate functions, while in ```mir_eval/onsets.py``` they're bundled together in ```f_measure```. What's your preference? I could put all 7 metrics in one big ```metrics``` function, or I could break them into 7 functions. If the latter, I'd definitely want to pass ```true_positives``` as an input for efficiency's sake.